recipe/build.sh

```diff
@@ -34,9 +59,9 @@ for core_config in "${CORES[@]}"; do
         "-DCMAKE_BUILD_TYPE=Release"
         "-DCMAKE_VERBOSE_MAKEFILE=ON"
         "$CORES_PATH"
+        "${CONF_FLAGS_FOR_CMAKE[@]}"
```

Comment:
```suggestion
       "${CMAKE_FLAGS}"
```
Does this work?  I think the flags you had to set are supposed to be set automatically by the `${CMAKE_FLAGS}` environment variable.

Reply:
I think the issue I had was that "-DHOST_ARCH=aarch64" is being set to i386 somewhere in the CMakeLists.txt, but I don't think I tried the CMAKE_ARGS, good catch. I'll cleanup if it passes. Thanks for the quick reply.

I plan to work on the -cli multi-arch after this goes through

Reply:
@danielnachun The CMAKE_ARGS does not set HOST_ARCH, nor the OSX_ARCHITECTURE for osx-arm64. I am reverting the change

Reply:
Okay, I'm going to push a commit with some small refactorings here to simplify logic a bit more, but now we know that these flags must be set explicitly.